vts/trustedservices/trustedservices_grpc.go

```diff
@@ -553,6 +620,158 @@ func (o *GRPC) GetEARSigningPublicKey(context.Context, *emptypb.Empty) (*proto.P
+		return nil, err
+	}
+
+	err = key.Set("alg", alg.String())
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := json.Marshal(key)
+	if err != nil {
+		return nil, err
+	}
+
+	bstring := string(b)
+
+	return &proto.PublicKey{
+		Key: bstring,
+	}, nil
+}
+
+func getEndorsementsError(err error) *proto.EndorsementQueryOut {
+	return &proto.EndorsementQueryOut{
+		Status: &proto.Status{
+			Result:      false,
+			ErrorDetail: fmt.Sprintf("%v", err),
+		},
+	}
+}
+
+func (o *GRPC) getEndorsementsFromStores(query *proto.EndorsementQueryIn) ([]byte, error) {
+	var q coserv.Coserv
+	if err := q.FromBase64Url(query.Query); err != nil {
+		return nil, err
+	}
+
+	// select store based on the requested artefact type
+	var (
+		store     kvstore.IKVStore
+		storeName string
+	)
+	switch q.Query.ArtifactType {
+	case coserv.ArtifactTypeEndorsedValues:
+		return nil, errors.New("endorsed value queries are not supported")
+	case coserv.ArtifactTypeReferenceValues:
+		storeName = "reference-value"
+		store = o.EnStore
+	case coserv.ArtifactTypeTrustAnchors:
+		storeName = "trust-anchors"
+		store = o.TaStore
+	}
+
+	profile, err := q.Profile.Get()
+	if err != nil {
+		return nil, err
+	}
+
+	// Look up a matching endorsement plugin
+	endorsementHandler, err := o.EndPluginManager.LookupByMediaType(fmt.Sprintf(`application/rim+cbor; profile=%q`, profile))
+	if err != nil {
+		return nil, err
+	}
+
+	scheme := endorsementHandler.GetAttestationScheme()
+
+	storeHandler, err := o.StorePluginManager.LookupByAttestationScheme(scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	queryKeys, err := storeHandler.SynthCoservQueryKeys(DummyTenantID, query.Query)
+	if err != nil {
+		return nil, err
+	}
+
+	var resultSet []string
+	for _, key := range queryKeys {
+		res, err := store.Get(key)
+		if err != nil && !errors.Is(err, kvstore.ErrKeyNotFound) {
+			return nil, fmt.Errorf("lookup %q in %s failed: %w", key, storeName, err)
+		}
+
+		resultSet = append(resultSet, res...)
+	}
+
+	coservResults, err := endorsementHandler.CoservRepackage(query.Query, resultSet)
+	if err != nil {
+		return nil, err
+	}
+
+	return coservResults, nil
+}
+
+func (o *GRPC) getEndorsementsFromProxy(handlerPlugin handler.ICoservProxyHandler, query *proto.EndorsementQueryIn) ([]byte, error) {
+	return handlerPlugin.GetEndorsements(DummyTenantID, query.Query)
+}
+
+func (o *GRPC) GetEndorsements(ctx context.Context, query *proto.EndorsementQueryIn) (*proto.EndorsementQueryOut, error) {
+	o.logger.Debugw("GetEndorsements", "media-type", query.MediaType)
+
+	var (
+		err           error
+		out           []byte
+		handlerPlugin handler.ICoservProxyHandler
+	)
+
+	// First, check to see if we have a CoSERV proxy plugin that can handle this query
+	handlerPlugin, err = o.CoservProxyPluginManager.LookupByMediaType(query.MediaType)
+	if err == nil {
+		// No error means we have a proxy plugin, so delegate to that.
+		out, err = o.getEndorsementsFromProxy(handlerPlugin, query)
+	} else {
```

Comment:
A more correct implementation is to check for a specific error that leads to conclusion that there is no suitable proxy plugin

Then, only cater to this error, other wise blind assumption can lead to problems down the line!